packages/ui-extensions/src/types/Button.ts

```diff
@@ -0,0 +1,58 @@
+export interface BaseButtonProps {
+  /**
+   * Destination to link to, renders a Link
+   */
+  to?: string;
+  /**
+   * The type of button that will be rendered.
+   *
+   * `primary`: button used for main actions or green-path. Ex: Continue to next step, Discount field
+   * `secondary`: button used for secondary actions not blocking user progress. Ex: Download Shop app
+   * `plain`: Renders a button that visually looks like a Link
+   * @default 'primary'*/
+  kind?: 'primary' | 'secondary' | 'plain';
+  /**
+   *  Specifies the color of the Button. The button will keep the style of the chodsen `kind`,
+   *  but replace its color according to the appearance.
+   *
+   * `monochrome`: button will inherit the color of its parent
+   * `critical`: button will take inherit the color of the Critical color group (in Checkout)
+   *  		 and map to 'destructive' (in Admin/Polaris). Typically used for destructive actions.
+   *
+   *  Excludes 'success' | 'warning' since buttons in a Banner will be `monochrome`
+   *  and use the default colors set by the Banner.
+   */
+  appearance?: 'monochrome' | 'critical';
+  /** Changes the size of the button
+   * @default 'base'
+   */
+  size?: 'base' | 'large';
+  /**
+   * Whether the button should fill all available inline space.
+   * */
+  fill?: boolean;
+  /**
+   * Replaces content with a loading indicator
+   */
+  loading?: boolean;
+  /**
+   * A label replacing the loading indicator when user prefers reduced motion
+   */
+  loadingLabel?: string;
```

Comment:
This API is also a little awkward when we have a separate `loading` prop, I think we might look at consolidating the label as an optional property on an object you can pass to `loading` in Checkout... cc/ @jffortier

Reply:
TBH, this option is a nice touch, but I've been close to removing it multiple times. I leaned towards the safe side when I introduce this difference for user that prefers reduced motion. The spinner is such a small motion in the overall UI and I couldn't find any validation that it would cause any harms.

I'd be ok to remove it OR the object route is elegant too. Is this what you had in mind Chris?
`loading?: boolean | { status: boolean; accessibilityLabel?: string; }`